x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java

```diff
@@ -817,6 +853,38 @@ public void testValidateFrozenPhaseHasSearchableSnapshot() {
         }
     }
 
+    public void testTimeseriesRegistryContainsAllActionsVersions() {
+        for (int version = INSTANCE.getLatestActionsOrderVersion(); version >= VERSION_ONE; version--) {
+            assertThat(ORDERED_VALID_HOT_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_WARM_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_COLD_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_FROZEN_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_DELETE_ACTIONS.get(version), notNullValue());
+        }
+    }
+
+    public void testTimeseriesRegistryDoesntRemoveActionsInFutureVersions() {
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_HOT_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_WARM_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_COLD_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_FROZEN_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_DELETE_ACTIONS);
+    }
+
+    private static void assertHigerVersionsContainActionsInLowerVersions(Map<Integer, List<String>> versionedActions) {
```

Comment:
Typo in `assertHigerVersionsContainActionsInLowerVersions` -> `assertHigherVersionsContainActionsInLowerVersions`

Reply:
++ Thanks Mary

Is the rest ... LGTM ? :)